packages/common/src/utils/oauth.ts

```diff
@@ -238,6 +328,54 @@ export interface OAuthRedirectParams {
     message: string;
 }
 
+// Scope descriptions mapping
+const SCOPE_DESCRIPTIONS: Record<string, OAuthScopeDescription> = {
+    read: {
+        title: 'Read access',
+        description:
+            'View your projects, dashboards, charts, and other resources',
+    },
+    write: {
+        title: 'Write access',
+        description:
+            'Create, update, and delete projects, dashboards, charts, and other resources',
+    },
+    'mcp:read': {
+        title: 'MCP read access',
+        description: 'Read access through Model Context Protocol',
+    },
+    'mcp:write': {
+        title: 'MCP write access',
+        description: 'Write access through Model Context Protocol',
+    },
+};
+
+/**
+ * Parses a scope string (e.g. "read write") and returns an array of scope objects with descriptions
+ */
+export const parseScopeString = (
+    scopeString: string,
+): OAuthScopeDescription[] => {
+    if (!scopeString || typeof scopeString !== 'string') {
+        return [];
+    }
+
+    const scopes = scopeString.trim().split(/\s+/);
+    return scopes
+        .map((scope) => SCOPE_DESCRIPTIONS[scope])
+        .filter((scope): scope is OAuthScopeDescription => scope !== undefined);
+};
+
+/**
+ * Gets a friendly client name from the client_id
+ */
+export const getClientName = (clientId: string): string => {
+    const clientNames: Record<string, string> = {
+        'lightdash-cli': 'Lightdash CLI tool',
+    };
+    return clientNames[clientId] || clientId;
```

Comment:
this looks like an overcomplicated way of formatting a single string... I'd used a switch instead.